### ISSUE TITLE
Use xref target heading instead of first heading as link title

### ISSUE THREAD

**Issue by KristianWedberg** [dotnet/docfx#2657]
**Feature request**

xref can link to the middle of a document using #. Documentation says:

> Title is extracted from the first heading of the Markdown file. Or you can also specify title
> using title metadata.

When linking to the middle of a document, I almost always want to use the _target_ heading as the title, and almost never the _first_ heading, as is the current default. I therefore have to specify the link text explicitly every time, and also keep the link text in sync with the target heading, which is onerous and error prone.

**Expected Behavior**:

If an xref links to the _middle_ of a document, use the _target_ heading text has the link text.

**Actual Behavior**:

If an xref links to the _middle_ of a document, the _first_ heading text is used as the link text.

**Comment by superyyrrzz** [comment]
Yes, this can be an improvement. If a UID is declared for a conceptual file, we can assign UIDs for every bookmark in it. The UID is like `file_uid#bookmark`. 
However, there is an issue that when using `<xref:file_uid#bookmark>`, `file_uid#bookmark` will not be taken as a whole UID. `#bookmark` will be reserved for bookmark use.

**Comment by herohua** [comment]
+ @live1206 for awareness
